src/Analyzers/CSharp/CodeFixes/MakePropertyRequired/CSharpMakePropertyRequiredCodeFixProvider.cs

```diff
@@ -0,0 +1,108 @@
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.MakePropertyRequired;
+
+[ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.MakePropertyRequired), Shared]
+internal sealed class CSharpMakePropertyRequiredCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+{
+    private const string CS8618 = nameof(CS8618); // Non-nullable variable must contain a non-null value when exiting constructor. Consider declaring it as nullable.
+
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public CSharpMakePropertyRequiredCodeFixProvider()
+    {
+    }
+
+    public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+        ImmutableArray.Create(CS8618);
+
+    public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        var document = context.Document;
+        var span = context.Span;
+        var cancellationToken = context.CancellationToken;
+
+        var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+        // Required members are available in C# 11 or higher
+        if (root.GetLanguageVersion() < LanguageVersion.CSharp11)
+            return;
+
+        var node = root.FindNode(span);
+
+        if (node is not PropertyDeclarationSyntax propertyDeclaration)
+            return;
+
+        var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+        var compilation = semanticModel.Compilation;
+
+        // `RequiredMember` attribute must be present in the metadata in order to emit required members
+        if (compilation.RequiredMemberAttributeType() is null)
+            return;
+
+        var propertySymbol = semanticModel.GetDeclaredSymbol(propertyDeclaration, cancellationToken);
+
+        if (propertySymbol is null)
+            return;
+
+        var setMethod = propertySymbol.SetMethod;
+
+        // Property must have a `set` or `init` accessor in order to be able to be required
+        if (setMethod is null)
+            return;
+
+        var containingTypeVisibility = propertySymbol.ContainingType.GetResultantVisibility();
+
+        var propertyAccessibility = propertySymbol.DeclaredAccessibility;
+        var setMethodAccessibility = setMethod.DeclaredAccessibility;
```

Comment:
I'm thinking if we can consolidate these two accessibilities into one with `(Accessibility)Math.Min((int)propertySymbol.DeclaredAccessibility, (int)setMethod.DeclaredAccessibility)` will that work?